src/sdk/sdk.ts

```diff
@@ -79,20 +79,27 @@ export async function writeEvent(
 
 /**
  * Get the owner reference for a custom resource
- * @param cr the custom resource to get the owner reference for
- * @returns the owner reference for the custom resource
+ * @param customResource the custom resource to get the owner reference for
+ * @returns the owner reference array for the custom resource
  */
-export function getOwnerRefFrom(cr: GenericKind): V1OwnerReference[] {
-  const { name, uid } = cr.metadata!;
+export function getOwnerRefFrom(customResource: GenericKind): V1OwnerReference[] {
+  const { name, uid } = customResource.metadata!;
 
-  return [
-    {
-      apiVersion: cr.apiVersion!,
-      kind: cr.kind!,
-      uid: uid!,
-      name: name!,
-    },
-  ];
+  const ownerReference: V1OwnerReference = {
+    apiVersion: customResource.apiVersion!,
+    kind: customResource.kind!,
+    uid: uid!,
+    name: name!,
+  };
+
+  if (Object.prototype.hasOwnProperty.call(customResource, "blockOwnerDeletion")) {
```

Comment:
This could be `hasOwn()`

Reply:
Went with this implementation based off linter feedback: https://eslint.org/docs/latest/rules/no-prototype-builtins

Reply:
a customResource does not contains these fields at all

```plaintext
KIND:       Pod
VERSION:    v1

DESCRIPTION:
    Pod is a collection of containers that can run on a host. This resource is
    created by clients and scheduled onto hosts.
    
FIELDS:
  apiVersion	<string>
    APIVersion defines the versioned schema of this representation of an object.
    Servers should convert recognized schemas to the latest internal value, and
    may reject unrecognized values. More info:
    https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources

  kind	<string>
    Kind is a string value representing the REST resource this object
    represents. Servers may infer this from the endpoint the client submits
    requests to. Cannot be updated. In CamelCase. More info:
    https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds

  metadata	<ObjectMeta>
    Standard object's metadata. More info:
    https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata

  spec	<PodSpec>
    Specification of the desired behavior of the pod. More info:
    https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status

  status	<PodStatus>
    Most recently observed status of the pod. This data may not be up to date.
    Populated by the system. Read-only. More info:
    https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
```

We have to place them on the CR
```plaintext
 > k explain pod.metadata.ownerReferences
KIND:       Pod
VERSION:    v1

FIELD: ownerReferences <[]OwnerReference>

DESCRIPTION:
    List of objects depended by this object. If ALL objects in the list have
    been deleted, this object will be garbage collected. If this object is
    managed by a controller, then an entry in this list will point to this
    controller, with the controller field set to true. There cannot be more than
    one managing controller.
    OwnerReference contains enough information to let you identify an owning
    object. An owning object must be in the same namespace as the dependent, or
    be cluster-scoped, so there is no namespace field.
    
FIELDS:
  apiVersion	<string> -required-
    API version of the referent.

  blockOwnerDeletion	<boolean>
    If true, AND if the owner has the "foregroundDeletion" finalizer, then the
    owner cannot be deleted from the key-value store until this reference is
    removed. See
    https://kubernetes.io/docs/concepts/architecture/garbage-collection/#foreground-deletion
    for how the garbage collector interacts with this field and enforces the
    foreground deletion. Defaults to false. To set this field, a user needs
    "delete" permission of the owner, otherwise 422 (Unprocessable Entity) will
    be returned.

  controller	<boolean>
    If true, this reference points to the managing controller.

  kind	<string> -required-
    Kind of the referent. More info:
    https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds

  name	<string> -required-
    Name of the referent. More info:
    http://kubernetes.io/docs/user-guide/identifiers#names

  uid	<string> -required-
    UID of the referent. More info:
    http://kubernetes.io/docs/user-guide/identifiers#uids

```

Reply:
Great, that `k explain` comparison was super useful for my context! Tracking the intended use now.